### DISCUSSION TITLE
Does the Root Provider in NextJS effect caching, server side rendering, or have other performance deficits?

### DISCUSSION THREAD

**Discussion by robertmalicke** [heroui-inc/heroui#3527] [General]
Because the NextUiProvider uses 'use client', does including it in the root of a NextJS layout prevent server-side rendering, caching performances, or any other performance deficits? I thought that including root providers in NextJS was a no-no because your entire app becomes a client application and loses SSR performance.

**Comment by alexnguyennz** [comment]
Are you using the navigate prop or any other [provider features](https://nextui.org/docs/api-references/nextui-provider)? If not, you could remove use client or remove the provider since it doesn't actually seem needed, not entirely sure though.

**Reply by Eloquence98** [reply]
Using a provider to wrap the server component as children won't affect server-side rendering (SSR) or caching in Next.js. This is because Next.js can render server components if they are imported at build time and used as children
